### PR TITLE
Add iosxr_logging_global module

### PR DESCRIPTION
**Pull request by ashwini-mhatre**
##### ISSUE TYPE

- New Module Pull Request

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Comment by ashwini-mhatre** [comment]
Module | statements | missing | excluded | branches | partial | coverage
-- | -- | -- | -- | -- | -- | --
Total | 292 | 7 | 0 | 131 | 23 | 93%
plugins/module_utils/network/iosxr/argspec/logging_global/__init__.py | 0 | 0 | 0 | 0 | 0 | 100%
plugins/module_utils/network/iosxr/argspec/logging_global/logging_global.py | 6 | 1 | 0 | 0 | 0 | 83%
plugins/module_utils/network/iosxr/config/logging_global/__init__.py | 0 | 0 | 0 | 0 | 0 | 100%
plugins/module_utils/network/iosxr/config/logging_global/logging_global.py | 85 | 1 | 0 | 73 | 3 | 97%
plugins/module_utils/network/iosxr/facts/logging_global/__init__.py | 0 | 0 | 0 | 0 | 0 | 100%
plugins/module_utils/network/iosxr/facts/logging_global/logging_global.py | 36 | 1 | 0 | 10 | 1 | 96%
plugins/module_utils/network/iosxr/rm_templates/logging_global.py | 80 | 3 | 0 | 46 | 18 | 83%
plugins/modules/iosxr_logging_global.py | 14 | 1 | 0 | 2 | 1 | 88%
tests/unit/modules/network/iosxr/test_iosxr_logging_global.py | 71 | 0 | 0 | 0 | 0 | 100%

File: plugins/modules/iosxr_logging_global.py [hunk 1]
```diff
@@ -0,0 +1,403 @@
...
+            choices:
+              - alerts
+              - critical
+              - debugging
+              - emergencies
+              - errors
+              - informational
+              - notifications
+              - warning
+              - disable
```
Comment (NilashishC) [inline plugins/modules/iosxr_logging_global.py:100]: AFAIU, `disable` is not a severity. It turns on/off console logging. We can have a subkey called `state` with `choices: ["enabled", "disabled"]` to cover this.

File: plugins/modules/iosxr_logging_global.py [hunk 2]
```diff
@@ -0,0 +1,403 @@
...
+              context_correlation:
+                type: bool
+                description: Specify enable correlation on context.
+              reissue_nonbistate:
+                type: bool
+                description: Specify reissue of non-bistate alarms on parent clear.This option is allowed for the rules whose type is stateful.
+              reparent:
+                type: bool
+                description: Specify reparent of alarm on parent clear.This option is allowed for the rules whose type is stateful.
+          rule_set:
```
Comment (KB-perByte) [inline plugins/modules/iosxr_logging_global.py:136]: make it plural maybe '_rule_sets_'

File: plugins/modules/iosxr_logging_global.py (hunk 1 above)
Comment (ashwini-mhatre) [reply plugins/modules/iosxr_logging_global.py]: ack

File: plugins/modules/iosxr_logging_global.py (hunk 2 above)
Comment (ashwini-mhatre) [reply plugins/modules/iosxr_logging_global.py]: ack

**Review by NilashishC — APPROVED** [review]

**Pull request merged**